### PR TITLE
Managed Group API 2/2

### LINKED ISSUE #79: Managed groups
- [ ] Define API for managed groups (stubs are there)
- [ ] Implement managed groups

### LINKED ISSUE #234: Refactor create_commit
The function `create_commit` should support a list of references instead a list of values for the proposals.

### LINKED ISSUE #235: KeyPackageBundle: add rekey function
I'd expect this to be something like `let new_key_package_bundle = existing_key_package_bundle.rekey()`

_Originally posted by @franziskuskiefer in https://github.com/openmls/openmls/pull/226#r529358962_

### PR DESCRIPTION
**Pull request by raphaelrobert**
This PR fixes open issues; 

Fixes #234. This makes `create_commit` more flexible and efficient.
Fixes #235. No new functionality, just following up on a previously discussed change.

Finally, this PR should also fix #79.

### CODE REVIEW

File: src/key_packages/mod.rs [hunk 1]
```diff
@@ -211,6 +211,23 @@ impl KeyPackage {
     pub(crate) fn cipher_suite(&self) -> &Ciphersuite {
         self.cipher_suite
     }
+
+    /// Replace the init key in the `KeyPackage` with a random one and return a
+    /// `KeyPackageBundle` with the corresponding secret values
+    pub fn rekey(&self) -> KeyPackageBundle {
```
Comment (kkohbrok) [inline src/key_packages/mod.rs:217]: This is a bit of a weird construction. I think it's ok for now, but with #214, we should really update the `KeyPackageBundle` directly, not the `KeyPackage`.

**Review by kkohbrok — APPROVED** [review]

File: src/key_packages/mod.rs (hunk 1 above)
Comment (franziskuskiefer) [reply src/key_packages/mod.rs:217]: Yeah I think we should just start tackling #214. This pops up quite a bit lately.

File: src/key_packages/mod.rs [hunk 2]
```diff
@@ -211,6 +211,23 @@ impl KeyPackage {
...
+        let ciphersuite = self.cipher_suite();
+        // Generate a new leaf secret and derive the key pair
+        let leaf_secret = Secret::from(get_random_vec(ciphersuite.hash_length()));
```
Comment (franziskuskiefer) [inline src/key_packages/mod.rs:220]: Couldn't you use `Secret::random` here?

File: src/key_packages/mod.rs [hunk 3]
```diff
@@ -211,6 +211,23 @@ impl KeyPackage {
...
+        let leaf_node_secret = KeyPackageBundle::derive_leaf_node_secret(ciphersuite, &leaf_secret);
```
Comment (franziskuskiefer) [inline src/key_packages/mod.rs:221]: I'd actually move all this into the `KeyPackageBundle` and have a `random` function there.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Yay, looks much cleaner.
Only one small thing.

File: src/key_packages/mod.rs (hunk 3 above)
Comment (raphaelrobert) [reply src/key_packages/mod.rs:221]: 😆 
That's where it was. Anyway, I moved it back now and fixed the `random` part.
I think it's better there anyway.

**Review by franziskuskiefer — APPROVED** [review]
Thanks, lgtm

**Pull request merged**